File: 94-BiTreeInOrderTrave/iterative.java
class Solution {
    private List<Integer> res;
    private List<TreeNode> stack;

    public List<Integer> inorderTraversal(TreeNode root) {
        res = new ArrayList<>();
        if (root == null) return res;
        
        stack = new LinkedList<>();
        TreeNode cur = root;
        while (cur != null || !stack.isEmpty()) {
            while (cur != null) {
                stack.add(cur);
                cur = cur.left;
            }
            cur = stack.remove(stack.size() - 1);
            res.add(cur.val);
            cur = cur.right;
        }
        return res;
    }
}

public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
}

File: 235-LowestCommonAncestorOfBST/iterative.cpp
#include <iostream>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    TreeNode* lowestCommonAncestor(TreeNode* root, TreeNode* p, TreeNode* q) {
        int left = p->val;
        int right = q->val;

        TreeNode* node = root;
        while (node) {
            int temp = node->val;
            if (left > temp && right > temp)
                node = node->right;
            else if (left < temp && right < temp)
                node = node->left;
            else return node;
        }
        return nullptr;
    }
};